tests/unittests/distros/test_init.py

```diff
@@ -306,13 +316,23 @@ def test_non_default_package_manager_fail(
             "cloudinit.distros.package_management.snap.Snap.install_packages",
             return_value=["pkg3"],
         )
-        _get_distro("debian").install_packages(
-            [{"apt": ["pkg1"]}, "pkg2", {"snap": ["pkg3"]}]
-        )
+        with pytest.raises(InstallerError):
```

Comment:
Let's add a match here that would have caught the failure in InstallerError message initialization.
```suggestion
        with pytest.raises(
            InstallerError,
            match="Failed to install the following packages: \['pkg3'\]",
        ):
```